src/commands/data/upsert/bulk.ts

```diff
@@ -34,6 +36,28 @@ export default class Upsert extends BulkOperationCommand {
     const conn = flags['target-org'].getConnection(flags['api-version']);
 
     await validateSobjectType(flags.sobject, conn);
+    // if the file is >150 mb, break it up into chunks, we'll deploy those individually
+    const fileStat = fs.statSync(flags.file);
+    if (fileStat.size >= 150000000) {
+      const batches = writeBatches(fs.readFileSync(flags.file, { encoding: 'utf8' }));
+
+      const results = await Promise.all(
+        batches.map(async (batch, i) =>
+          this.runBulkOperation(flags.sobject, `batch${i}.csv`, conn, flags.async ? 0 : flags.wait?.minutes, 'upsert', {
```

Comment:
nit: don't need to compute wait time with each batch.  It's the same for each one so we can compute outside of the loop.  We can also reuse it for the < 150MB code path.